doc/doxygen/public/coding-conventions.doxygen

```diff
@@ -0,0 +1,573 @@
+
+  Native line endings should be allowed on each platform. This is desirable as Visual Studio for example will always insert <code>CRLF</code> even if the file is <code>LF</code> only, leading to a mixed line ending style for this file. Native <code>eol</code> style avoids this problem.
+
+  @subsection bracket_placements Bracket placements
+
+  Matching pairs of opening and closing curly braces should be set to the same column. See the following example:
+
+  @code{cpp}
+    while (continue == true)
+    {
+      for (int i = 0; i < 10; i++)
+      {
+        ...
+      }
+      if (x < 7)
+      {
+        ....
+      }
+    }
+  @endcode
+
+  The main reason for this rule is to avoid constructions like:
+
+  @code{cpp}
+    if (isValid(a))
+      error = 0;
+      return 0;
+  @endcode
+
+  that might later be changed to something like:
+
+  @code{cpp}
+    if (isValid(a))
+      error = 0;
+      return 0;
+  @endcode
+
+  Use braces around a block even if you use only single line. Single line constructs for trivial tests like:
+
+  @code{cpp}
+    if (test) continue;
+  @endcode
+
+  are allowed.
+
+  One exception is several if/else statements, which can be written as:
+
+  @code{cpp}
+    if (cond1)
+    {
+      ...
+    }
+    else if (cond2)
+    {
+      ...
+    }
+    else if (cond3)
+    {
+      ...
+    }
+    else
+    {
+      ...
+    }
+  @endcode
+
+  This is safe because the first statement in each else branch is used, which is itself braced by the if branch.
+
+  @section naming_conventions Naming conventions
+
+  The following section describes the naming conventions followed by OpenMS developers.
+
+  @subsection reserved_words Reserved words
+
+  Reserved words of the C++ language and symbols defined e.g. in the <code>STL</code> or in the standard C library must not be used as names for classes or class members. Examples include but are not limited to: <code>set</code>, <code>map</code>, <code>exp</code> and <code>log</code>.
+
+  @subsection file_names File names
+
+  Header files and source files should have the same name as the classes they contain. Source files end in <code>.cpp</code>, while header files end in <code>.h</code>. File names should be capitalised exactly as the class they contain (see below). Each header/source file should contain one class only, although exceptions are possible for light-weight classes.
+
+  @subsection underscores Underscores
+
+  The usage of underscores in names has two different meanings: A trailing "_" at the end indicates that something is protected or private to a class. Apart from that, different parts of a name are sometimes separated by an underscore, and sometimes separated by capital letters.
+
+  @note According to the C++ standard, names that start with an underscore are reserved for internal purposes of the language and its standard library, so you should never use them.
+
+  @subsection classes_types_namespaces Classes/Types/Namespaces
```

Comment:
does this hash? say, `page-url.something#Classes/Types/Namespaces`, i think `/` would change to some code.